gitlab/resource_gitlab_project_push_rules.go

```diff
@@ -98,7 +98,17 @@ func resourceGitlabProjectPushRulesCreate(d *schema.ResourceData, meta interface
 
 	pushRules, _, err := client.Projects.AddProjectPushRule(project, options)
 	if err != nil {
-		return err
+		// Delete existing push rules
+		_, err := client.Projects.DeleteProjectPushRule(project)
```

Comment:
It would be better to use the EditProjectPushRule API method, especially since some of the issues mention rules getting automatically recreated when they try to delete push rules manually.